lib/streamlit/elements/selectbox.py

```diff
@@ -204,19 +206,20 @@ def _selectbox(
 
         opt = ensure_indexable(options)
 
-        if not isinstance(index, int):
+        clearable = index is None
+        if not isinstance(index, int) and not clearable:
```

Comment:
nit: I think `if not isinstance(index, int) and index is not None` would be slightly more future proof in case we add clearable as a parameter in the future.